boards/nrf52dk/src/main.rs

```diff
@@ -311,24 +311,28 @@ pub unsafe fn reset_handler() {
     kernel::debug::assign_console_driver(Some(console), kc);
 
     let ble_radio = static_init!(
-        capsules::ble_advertising_driver::BLE<
+        nrf52::ble_connection::ble_advertising_driver::BLE<
             'static,
-            nrf52::radio::Radio,
+            nrf52::ble_connection::radio::Radio,
             VirtualMuxAlarm<'static, Rtc>,
         >,
-        capsules::ble_advertising_driver::BLE::new(
-            &mut nrf52::radio::RADIO,
+        nrf52::ble_connection::ble_advertising_driver::BLE::new(
+            &mut nrf52::ble_connection::radio::RADIO,
             kernel::Grant::create(),
-            &mut capsules::ble_advertising_driver::BUF,
+            &mut nrf52::ble_connection::ble_advertising_driver::BUF,
             ble_radio_virtual_alarm
         )
```

Comment:
hrm, can't we not have both `ble_drivers` enabled here? It's not a long-term solution but I'm not sure if it `your` modified syscall implementation works with the changes in `libtock`?